packages/rgbpp/src/rgbpp/utils/transaction.ts

```diff
@@ -0,0 +1,61 @@
+import { BaseCkbVirtualTxResult } from '@rgbpp-sdk/ckb';
+import { BtcAssetsApi } from '@rgbpp-sdk/service';
+
+export interface RgbppTxGroup {
+  ckbVirtualTxResult: BaseCkbVirtualTxResult | string;
+  btcTxHex: string;
+}
+
+export interface SentRgbppTxGroup {
+  btc: {
+    sent: boolean;
+    txId?: string;
+    error?: Error | unknown;
+  };
+  ckb: {
+    sent: boolean;
+    error?: Error | unknown;
+  };
+}
+
+export async function sendRgbppTxGroups(props: {
+  txGroups: RgbppTxGroup[];
+  btcService: BtcAssetsApi;
+  retry?: number;
+}): Promise<SentRgbppTxGroup[]> {
+  const results: SentRgbppTxGroup[] = [];
+  for (const group of props.txGroups) {
+    const result: SentRgbppTxGroup = {
+      ckb: {
+        sent: false,
+      },
+      btc: {
+        sent: false,
+      },
+    };
+    try {
+      // TODO: add retry logic
+      const sent = await props.btcService.sendBtcTransaction(group.btcTxHex);
+      result.btc.txId = sent.txid;
+      result.btc.sent = true;
+    } catch (e) {
+      result.btc.sent = false;
+      result.btc.error = e;
+    }
+    if (result.btc.sent) {
+      try {
+        await props.btcService.sendRgbppCkbTransaction({
```

Comment:
It is recommended to place `sendRgbppCkbTransaction` right after `sendBtcTransaction`, so there will be no need to maintain a status list of whether the transactions were sent successfully.

Reply:
Do you mean like this?
```ts
try {
  // TODO: add retry logic
  const sent = await props.btcService.sendBtcTransaction(group.btcTxHex);
  result.btc.txId = sent.txid;
  result.btc.sent = true;
  
  try {
    await props.btcService.sendRgbppCkbTransaction({
      btc_txid: result.btc.txId!,
      ckb_virtual_result: group.ckbVirtualTxResult,
    });
    result.ckb.sent = true;
  } catch (e) {
    result.ckb.sent = false;
    result.ckb.error = e;
  }
} catch (e) {
  result.btc.sent = false;
  result.btc.error = e;
}
```

Reply:
Yes. 

I think `result.btc.sent` is not needed, if the BTC `txId` or `result.btc.error` is not null, the BTC transaction must be committed successfully, and the `result.ckb.sent` is not needed, either.

Reply:
The `result.x.sent` props have removed, updated at 37074c4. Tho I'm not wrapping try-catch blocks due to readability reasons.